spinalcordtoolbox/testing/create_test_data.py

```diff
@@ -204,3 +205,33 @@ def dummy_segmentation(size_arr=(256, 256, 256), pixdim=(1, 1, 1), dtype=np.floa
     if debug:
         img.save('tmp_dummy_seg_'+datetime.now().strftime("%Y%m%d%H%M%S%f")+'.nii.gz')
     return img
+
+def dummy_4d(vol_num=10, size_arr=(256, 256, 256), pixdim=(1, 1, 1), dtype=np.float64, orientation='LPI',
+             shape='rectangle', angle_RL=0, angle_AP=0, angle_IS=0, radius_RL=5.0, radius_AP=3.0,
+             interleaved=False, zeroslice=[], debug=False):
+    """
+    Create a dummy 4D segmentation (dMRI/fMRI)
+    :param vol_num: int: number of volumes in 4D data
+    :return: Image object
+    """
+
+    from random import uniform
+    from sct_image import concat_data
+
+    img_list = []
+
+    # Loop across individual volumes of 4D data
+    for volume in range(0,vol_num):
+        factor = uniform(0.5, 3.0)          # shift in voxels
```

Comment:
This factor could probably takes into account breath frequency and repetition time in some way.

Reply:
it could, but i would not worry too much (as TR can also vary by several seconds)